python/tvm/relax/transform/legalize_ops/manipulate.py

```diff
@@ -206,8 +212,10 @@ def te_layout_transform(data, name):
     # Create TIR schedule to apply layout changes with axis separators
     sch = tir.Schedule(tir_func)
     sch.transform_layout(primfunc_name, ("write", 0), index_map, pad_value)
-    if len(axis_separators) != 0:
-        sch.set_axis_separator(primfunc_name, ("write", 0), axis_separators=axis_separators)
+    set_axis_sep(axis_separators, sch, "write")
+    if input_axis_separators is not None:
+        input_axis_separators = list(map(lambda x: x.value, input_axis_separators))
```

Comment:
It turns out that this change is needed because `set_axis_separator` directive expects a list: https://ci.tlcpack.ai/blue/organizations/jenkins/tvm-unity/detail/PR-17115/7/pipeline/
`TypeError: In "Schedule.set_axis_separator", "axis_separators" has wrong type. Expected "list", but gets: "Array"`

The initial setting of `list` for `axis_separators` is also needed to name `prim_func`. Hence, I have retained it. 

however, I've updated the way in which `list` is getting created. `axis_sep = [int(sep) for sep in axis_sep]`

Reply:
Good catch, and looks like the error message is caused by the `@type_checked` decorator on `Schedule.set_axis_separator`, which doesn't recognize an `Array<IntImm>` as equivalent to `List[int]`.  Since the FFI handles type-checking internally, including any type coercions, the `@type_checked` decorator probably should be removed altogether.  But that's a later cleanup for me to do.

For this PR, the update to `axis_sep = [int(sep) for sep in axis_sep]` looks good,  thank you!